### DISCUSSION TITLE
VroidSDKで出来ることなどについての質問

### DISCUSSION THREAD

**Discussion by nira-yusan** [pixiv/vroid-sdk-developers#111] [Q&A]
はじめましてです。VroidSDKの使用を検討しているのですが、それにあたっていくつか知りたいことがあります。

・VroidHubから選択したVroidの情報は、スクリプトではどのようにして受けとることができますか？

・「一度選択したことのあるモデルをVroidHubから自動で読み込む」という機能を考えてるのですが、そういったことは可能でしょうか？また、自動で読み込むためになにかしらのアドレスなどといった必要な情報があるなら知りたいです。

・マルチプレイ用のSDKの使用も検討しているのですが、検索しても情報が少なくあまり把握できていません。仮にネット対戦などで相手の選択しているVroidを自分側の画面にも表示する、といったことはどうすればできますか？

**Comment by nyamadan** [comment]
> ・VroidHubから選択したVroidの情報は、スクリプトではどのようにして受けとることができますか？

基本的な手順は[README](https://github.com/pixiv/vroid-sdk-developers/blob/main/README.ja.md#%E3%82%B9%E3%82%AF%E3%83%AA%E3%83%97%E3%83%86%E3%82%A3%E3%83%B3%E3%82%B0)が参考になると思います。
たとえば `GetAccountCharacterModels` を使用するとログインしているアカウントでアップロードしたモデルの情報が取得できます。
取得できる情報は[ドキュメント](https://developer.vroid.com/sdk/docs/0.3.0/Pixiv.VroidSdk.Api.DataModel.CharacterModel.html)に記載があります。

> ・「一度選択したことのあるモデルをVroidHubから自動で読み込む」という機能を考えてるのですが、そういったことは可能でしょうか？また、自動で読み込むためになにかしらのアドレスなどといった必要な情報があるなら知りたいです。

VRoid SDKではそういった機能は提供しておりませんが[CharacterModel.id](https://developer.vroid.com/sdk/docs/0.3.0/Pixiv.VroidSdk.Api.DataModel.CharacterModel.html#Pixiv_VroidSdk_Api_DataModel_CharacterModel_id)を保存しておけば自動ロードは実装できそうです。

> ・マルチプレイ用のSDKの使用も検討しているのですが、検索しても情報が少なくあまり把握できていません。仮にネット対戦などで相手の選択しているVroidを自分側の画面にも表示する、といったことはどうすればできますか？

対戦相手の使用している3Dモデルを自分の側の画面に表示するといったことはマルチプレイのSDKを使用すれば可能となります。
具体的なコードについてはマルチプレイSDK同梱のExampleが参考になると思いますのでSDKの利用申請をいただければと思います

参考： #7

**Reply by nira-yusan** [reply]
なるほどです………
今、VroidSDKを利用しないゲームの開発に手を付けていて、まだ申請ができる状態ではないのでマルチプレイSDKについて聞ける範囲で聞きたいことがあります。Readmeを見たところvrmをダウンロードしてキャッシュとして保存するとありましたが、これはマルチプレイSDKで相手のモデルを読み込む時も似たような感じになるのですか？

**Reply by nyamadan** [reply]
> 今、VroidSDKを利用しないゲームの開発に手を付けていて、まだ申請ができる状態ではないのでマルチプレイSDKについて聞ける範囲で聞きたいことがあります。
> Readmeを見たところvrmをダウンロードしてキャッシュとして保存するとありましたが、これはマルチプレイSDKで相手のモデルを読み込む時も似たような感じになるのですか？

はい、通常のSDKと同様に自動的にキャッシュいたします。

マルチプレイに関する追加の情報として、対戦相手のモデルを ~~タウンロードしたければ~~ 読み込みたければ対戦相手のモデルのダウンロードIDを何らかの方法で取得する必要があり、これを対戦相手と交換する必要があります。

マルチプレイにおいてはReqQueueという仕組みを導入しており、複数モデルを優先度をつけて読み込むことが可能です。
https://vroid.notion.site/ReqQueue-4b0c0076ed934a11b26797bf07cdb56b

マルチプレイSDKの申請お待ちしております。

**Reply by nira-yusan** [reply]
なるほど、一時的に表示するだけでなく、ダウンロードも可能なんですね

**Reply by nyamadan** [reply]
> なるほど、一時的に表示するだけでなく、ダウンロードも可能なんですね

失礼しました、訂正いたします。
SDKでモデルのダウンロードはサポートしておりません。基本的には一時的に利用していただくものとなります。

**Reply by nira-yusan** [reply]
訂正された内容を読みましたが、これって相手のモデルを画面に表示するには、相手のモデルのダウンロードIDが必要……ということであっていますか？

**Reply by nyamadan** [reply]
はい。あっています。通常のSDKと仕組みが同じものと認識いただければと思います。

**Reply by nyamadan** [reply]
こちらもダウンロードIDと表現してしまいましたが[ダウンロードライセンスID](https://developer.vroid.com/sdk/docs/0.3.0/Pixiv.VroidSdk.Api.DataModel.DownloadLicense.html#Pixiv_VroidSdk_Api_DataModel_DownloadLicense_id)が正確な名称となります。

重ねて訂正いたします、申し訳ございません。

**Reply by nira-yusan** [reply]
となると、例えばランダムでマッチングした相手のモデルを表示(表示するだけ)するためには、モデルの規約がダウンロードライセンスIDの取得が許可されてる範囲である必要がありますか？

**Reply by nira-yusan** [reply]
すみません、追加でダウンロードライセンスIDについて質問したいのですが、これは一時的にモデルを取得するためのもの………という認識でいいのですか？

**Reply by nyamadan** [reply]
> となると、例えばランダムでマッチングした相手のモデルを表示(表示するだけ)するためには、モデルの規約がダウンロードライセンスIDの取得が許可されてる範囲である必要がありますか？

その通りです。例えば格闘ゲームを作ったとして、使用できるものは暴力表現が許可されているものに限ります。

> すみません、追加でダウンロードライセンスIDについて質問したいのですが、これは一時的にモデルを取得するためのもの………という認識でいいのですか？

はい、その通りです！

**Reply by nira-yusan** [reply]
ご回答ありがとうございます。
これとは別に聞きたいことがあるのですが、ローカライズが推奨されていますが、するかしないかでどれくらいプレイするユーザーに差がでるか知りたいです。回答できるのであれば是非聞きたいです！
